crates/tx-graph/src/transactions/assert_chain.rs

```diff
@@ -96,3 +104,91 @@ impl AssertChain {
         }
     }
 }
+
+/// This is needed because the blanket implementation of serde's deserializers doesn't go past fixed
+/// length arrays of larger than 32.
+pub fn serialize_assert_vector<T: Serialize, S: Serializer>(
+    data: &[T; NUM_ASSERT_DATA_TX],
+    serializer: S,
+) -> Result<S::Ok, S::Error> {
+    let mut seq = serializer.serialize_tuple(NUM_ASSERT_DATA_TX)?;
+    for e in data {
+        seq.serialize_element(e)?;
+    }
+    seq.end()
+}
+
+/// This is needed because the blanket implementation of serde's deserializers doesn't go past fixed
+/// length arrays of larger than 32.
+pub fn deserialize_assert_vector<'de, D: Deserializer<'de>, T: Deserialize<'de>>(
+    deserializer: D,
+) -> Result<[T; NUM_ASSERT_DATA_TX], D::Error> {
+    // THE AUTHORS OF SERDE CAN BURN IN HELL. Seriously whoever thought of serde's design is fucking
+    // retarded.
```

Comment:
Yes we're so back!

Reply:
We should hold ourselves to a more professional standard for comments.

Reply:
It could reasonably be argued, for instance, that this would violate our own [code of conduct](https://github.com/alpenlabs/strata-bridge/blob/main/CONTRIBUTING.md#code-of-conduct).